### PR TITLE
Add Electrostatic Force Calculator (GUI)

### PR DESCRIPTION
**Pull request by Priyanshu729**
THIS IS CLASS 12 ONE OF THE BEST PROJECT MADE BY ME . ONE MORE FILE IS NEEDED FOR RUN THIS PROGRAM i.e COMPUTE.PY
RUN BOTH IN SAME FILE AND WATCH HOW CHARGE IS MOVING IT'S AMAZING

### CODE REVIEW

**Comment by Priyanshu729** [comment]
Is it ok?? Is it amazed u??

File: python/electrostatic_force.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+                        distancenew=canvas.create_text(f,g,text=f"{distnew}")
+                        x_11=m
+                        y_11=n
+            elif m>730:#right side of the blank bar
+                a1=m-30
+                b1=n-30
+                a2=m+30
+                b2=n+30
+                try:  
+                    canvas.delete(text4)
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Using variable 'text4' before assignment](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

**Comment by tasnimzotder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 84
           

Complexity increasing per file
==============================
- python/electrostatic_force.py  31
         

Clones added
============
- python/electrostatic_force.py  45
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653&bid=20438243)

File: python/electrostatic_force.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+                    canvas.delete(text3)
+                    canvas.delete(oval3)
+                    canvas.delete(line3)
+                    canvas.delete(distancenew)
+                    oval3=canvas.create_oval(x1,y1,x2,y2,fill="orange")
+                    text3=canvas.create_text(m,n,text=f"{charge1val}")
+                    line3=canvas.create_line(x2,n,xaxis,yaxis,fill="black",width="2")
+                    f,g=compute.mp(x2,n,xaxis,yaxis)
+                    distnew=int((compute.liner(x2,n,xaxis,yaxis))//80)
+                    distancenew=canvas.create_text(f,g,text=f"{distnew}")
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+entry1=Entry(root,textvariable=Charge1,bd=3,relief="sunken")
+entry1.place(x=190,y=26)
+Charge2=IntVar()
+Charge2.set(1)
+entry2=Entry(root,textvariable=Charge2,bd=3,relief="sunken")
+entry2.place(x=190,y=66)
+box=IntVar()
+label3=Label(root,text="MEDIUM :",font="TIMESNEWROMAN 15 bold",bg="grey",fg="black",bd=2,relief="solid")
+label3.place(x=100,y=100)
+box1=Radiobutton(root,text="Air",value=1,variable=box,bd=2,relief="sunken")
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['Radiobutton' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+                    y_axis=250
+                    flagsmall=0         
+            if m<650:    
+                x1=m-50
+                y1=n-50
+                x2=m+50
+                y2=n+50
+                try:
+                    canvas.delete(text3)
+                    canvas.delete(oval3)
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Using variable 'oval3' before assignment](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+        oval2=canvas.create_oval(a_1,b_1,a_2,b_2,fill="grey")
+        text1=canvas.create_text(600,250,text=f"{charge1val}",fill="white")
+        text2=canvas.create_text(760,250,text=f"{charge2val}",fill="white")
+        line1=canvas.create_line(650,250,730,250,fill="black",width="3")
+        countbigclick=0
+        countsmallclick=0
+        f,g=compute.mp(650,250,730,250)
+        dist=int((compute.liner(650,250,730,250)//80))
+        distance=canvas.create_text(f,g,text=f"{dist}") 
+        flag=1
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+r1=Radiobutton(root,text="pm",value=123,variable=i,bd=2,relief="sunken")
+r2=Radiobutton(root,text="cm",value=234,variable=i,bd=2,relief="sunken")
+r3=Radiobutton(root,text="mm",value=345,variable=i,bd=2,relief="sunken")
+r4=Radiobutton(root,text="nm",value=456,variable=i,bd=2,relief="sunken")
+r1.place(x=200,y=150)
+r2.place(x=260,y=150)
+r3.place(x=320,y=150)
+r4.place(x=380,y=150)
+mango=IntVar()
+label5=Label(root,text="CHARGE'S UNIT:",font="TIMESNEWROMAN 12 bold",bg="grey",fg="black",bd=2,relief="solid").place(x=50,y=200)
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['Label' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+    root1.minsize(500,600)
+    canvas=Canvas(root1,width=2000,height=2200,bg="sky blue")
+    canvas.place(x=0,y=0)
+    M=StringVar()
+    x_1,y_1,x_2,y_2=550,200,650,300  #coordinates for the big circle
+    a_1,b_1,a_2,b_2=730,220,790,280  #coordinates for the small circle
+    global countsmallclick,countbigclick
+    #global bigclick,smallclick
+    bigclick=0
+    smallclick=0
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [local variable 'smallclick' is assigned to but never used (F841)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+                    x_axis=650     #coordinates for smaller circle
+                    y_axis=250
+                    flagsmall=0         
+            if m<650:    
+                x1=m-50
+                y1=n-50
+                x2=m+50
+                y2=n+50
+                try:
+                    canvas.delete(text3)
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Using variable 'text3' before assignment](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+    canvas.bind("<Button-1>",dad)                        
+    root1.mainloop()
+#second window ka mainloop hai confuse mat hona  
+############################################################################################################################################################
+#FIRST WINDOW BY AWASTHIJI 
+root=Tk()
+root.title("COULOMB'S LAW")
+root.minsize(550,350)
+root.maxsize(550,350)
+label1=Label(root,text="Charge1",font="TIMESNEWROMAN 15 bold",bg="grey",fg="black",bd=2,relief="solid")
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['Label' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+    force=compute.forcecalc(charge1val,charge2val,q,dist,bumbam)
+    M.set(str(force))
+    lavel=Label(root1,textvariable=M,font="TIMESNEWROMAN 12 bold",bd=3,height=1,fg="green") 
+    lavel.place(x=865,y=600)
+    lav=Label(root1,text="The Electrostatic Force Of Attraction is :",font="TIMESNEWROMAN 12 bold",bg="white",fg="red")
+    lav.place(x=550,y=600)
+    def dad(event):#radius of bigcircle=50, radius of smaller circle=30
+        global x11,x_11
+        global y11,y_11
+        global lavel1,heaven
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Using global for 'lavel1' but no assignment is done](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 11]
```diff
@@ -0,0 +1,362 @@
...
+    root1.geometry("7000x6000")
+    root1.minsize(500,600)
+    canvas=Canvas(root1,width=2000,height=2200,bg="sky blue")
+    canvas.place(x=0,y=0)
+    M=StringVar()
+    x_1,y_1,x_2,y_2=550,200,650,300  #coordinates for the big circle
+    a_1,b_1,a_2,b_2=730,220,790,280  #coordinates for the small circle
+    global countsmallclick,countbigclick
+    #global bigclick,smallclick
+    bigclick=0
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Unused variable 'bigclick'](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 12]
```diff
@@ -0,0 +1,362 @@
...
+    #print(charge1val)
+    charge2val=Charge2.get()
+    #print(charge2val)
+    root.destroy()
+    root1=Tk()
+    global flag
+    root1.title("OUTPUT")
+    root1.geometry("7000x6000")
+    root1.minsize(500,600)
+    canvas=Canvas(root1,width=2000,height=2200,bg="sky blue")
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['Canvas' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 13]
```diff
@@ -0,0 +1,362 @@
...
+                    canvas.delete(line3)
+                    oval4=canvas.create_oval(a1,b1,a2,b2,fill="orange")
+                    text4=canvas.create_text(m,n,text=f"{charge2val}")
+                    line3=canvas.create_line(x_axis,y_axis,a1,n,fill="black",width="2")
+                    f,g=compute.mp(x_axis,y_axis,a1,n)
+                    distnew=int((compute.liner(x_axis,y_axis,a1,n))//60)
+                    distancenew=canvas.create_text(f,g,text=f"{distnew}")
+                    x11=m
+                    y11=n
+                except NameError or UnboundLocalError:
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 14]
```diff
@@ -0,0 +1,362 @@
...
+    #print(charge2val)
+    root.destroy()
+    root1=Tk()
+    global flag
+    root1.title("OUTPUT")
+    root1.geometry("7000x6000")
+    root1.minsize(500,600)
+    canvas=Canvas(root1,width=2000,height=2200,bg="sky blue")
+    canvas.place(x=0,y=0)
+    M=StringVar()
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['StringVar' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 15]
```diff
@@ -0,0 +1,362 @@
...
+    lavel=Label(root1,textvariable=M,font="TIMESNEWROMAN 12 bold",bd=3,height=1,fg="green") 
+    lavel.place(x=865,y=600)
+    lav=Label(root1,text="The Electrostatic Force Of Attraction is :",font="TIMESNEWROMAN 12 bold",bg="white",fg="red")
+    lav.place(x=550,y=600)
+    def dad(event):#radius of bigcircle=50, radius of smaller circle=30
+        global x11,x_11
+        global y11,y_11
+        global lavel1,heaven
+        global oval3,oval4
+        global text3,text4
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Global variable 'text3' undefined at the module level](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 16]
```diff
@@ -0,0 +1,362 @@
...
+        f,g=compute.mp(670,250,730,250)
+        dist=int((compute.liner(670,250,730,250))//60) 
+        distance=canvas.create_text(f,g,text=f"{dist}")
+        flag=2      
+    bumbam=i.get()
+    bambum=mango.get()
+    q=compute.getter(bambum)
+    force=compute.forcecalc(charge1val,charge2val,q,dist,bumbam)
+    M.set(str(force))
+    lavel=Label(root1,textvariable=M,font="TIMESNEWROMAN 12 bold",bd=3,height=1,fg="green")
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['Label' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 17]
```diff
@@ -0,0 +1,362 @@
...
+                    flagsmall=0         
+            if m<650:    
+                x1=m-50
+                y1=n-50
+                x2=m+50
+                y2=n+50
+                try:
+                    canvas.delete(text3)
+                    canvas.delete(oval3)
+                    canvas.delete(line3)
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Using variable 'line3' before assignment](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 18]
```diff
@@ -0,0 +1,362 @@
...
+box1=Radiobutton(root,text="Air",value=1,variable=box,bd=2,relief="sunken")
+box1.place(x=210,y=100)
+box4=Radiobutton(root,text="Vaccum",value=2,variable=box,bd=2,relief="sunken")
+box4.place(x=280,y=100)
+box5=Radiobutton(root,text="Free Space",value=3,variable=box,bd=2,relief="sunken")
+box5.place(x=380,y=100)
+label4=Label(root,text="UNIT :",font="TIMESNEWROMAN 16 bold",bg="grey",fg="black",bd=2,relief="solid")
+label4.place(x=100,y=150)
+i=IntVar()
+r1=Radiobutton(root,text="pm",value=123,variable=i,bd=2,relief="sunken")
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['Radiobutton' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 19]
```diff
@@ -0,0 +1,362 @@
...
+                        oval4=canvas.create_oval(a1,b1,a2,b2,fill="orange")
+                        text4=canvas.create_text(m,n,text=f"{charge2val}")
+                        line3=canvas.create_line(x_axis,y_axis,a1,n,fill="black",width="2")
+                        x11=m
+                        y11=n
+                        f,g=compute.mp(x_axis,y_axis,a1,n)
+                        distnew=int((compute.liner(x_axis,y_axis,a1,n))//60)
+                        distancenew=canvas.create_text(f,g,text=f"{distnew}")
+        bamchik=i.get()
+        bumchik=mango.get()
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [local variable 'bumchik' is assigned to but never used (F841)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 20]
```diff
@@ -0,0 +1,362 @@
...
+                        x_11=m
+                        y_11=n
+            elif m>730:#right side of the blank bar
+                a1=m-30
+                b1=n-30
+                a2=m+30
+                b2=n+30
+                try:  
+                    canvas.delete(text4)
+                    canvas.delete(oval4)
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Using variable 'oval4' before assignment](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 21]
```diff
@@ -0,0 +1,362 @@
...
+                        distnew=int((compute.liner(x2,n,xaxis,yaxis))//80)
+                        distancenew=canvas.create_text(f,g,text=f"{distnew}")
+                        x_11=m
+                        y_11=n
+            elif m>730:#right side of the blank bar
+                a1=m-30
+                b1=n-30
+                a2=m+30
+                b2=n+30
+                try:
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 22]
```diff
@@ -0,0 +1,362 @@
...
+        global x11,x_11
+        global y11,y_11
+        global lavel1,heaven
+        global oval3,oval4
+        global text3,text4
+        global line3,line4
+        global distancenew
+        heaven=0
+        m=event.x
+        n=event.y
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 23]
```diff
@@ -0,0 +1,362 @@
...
+root.title("COULOMB'S LAW")
+root.minsize(550,350)
+root.maxsize(550,350)
+label1=Label(root,text="Charge1",font="TIMESNEWROMAN 15 bold",bg="grey",fg="black",bd=2,relief="solid")
+label1.place(x=100,y=20)
+label2=Label(root,text="Charge2",font="TIMESNEWROMAN 15 bold",bg="grey",fg="black",bd=2,relief="solid")
+label2.place(x=100,y=60)
+Charge1=IntVar()
+Charge1.set(1)
+entry1=Entry(root,textvariable=Charge1,bd=3,relief="sunken")
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: ['Entry' may be undefined, or defined from star imports: tkinter (F405)](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 24]
```diff
@@ -0,0 +1,362 @@
...
+                    if flagsmall==0:  
+                        canvas.delete(distance)
+                        canvas.delete(line2)
+                        canvas.delete(oval2)
+                        canvas.delete(text2)
+                        oval4=canvas.create_oval(a1,b1,a2,b2,fill="orange")
+                        text4=canvas.create_text(m,n,text=f"{charge2val}")
+                        line3=canvas.create_line(x_axis,y_axis,a1,n,fill="black",width="2")
+                        x11=m
+                        y11=n
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

File: python/electrostatic_force.py [hunk 25]
```diff
@@ -0,0 +1,362 @@
...
+import compute
+def com1():
+    global lavel
+    charge1val=Charge1.get()
+    #print(charge1val)
+    charge2val=Charge2.get()
+    #print(charge2val)
+    root.destroy()
+    root1=Tk()
+    global flag
```
Comment (tasnimzotder) [inline python/electrostatic_force.py]: Codacy found an issue: [Global variable 'flag' undefined at the module level](https://app.codacy.com/gh/tasnimzotder/hacktoberfest-2020/pullRequest?prid=6279653)

**Comment by Priyanshu729** [comment]
where r u getting error .pls tell me?

**Comment by Priyanshu729** [comment]
It's a python based program use vs code or uske to run this

**Comment by Priyanshu729** [comment]
Run this in vs code if error comes then tell me

**Comment by tasnimzotder** [comment]
@Priyanshu729 It's really awesome

image

**Comment by tasnimzotder** [comment]
Mind-Blowing 🤯
image

**Review by tasnimzotder — APPROVED** [review]
🎉

**Comment by Priyanshu729** [comment]
How is it looks is it amazed u??
Bro run similarly the sqlite and tkinter program.It was also amazed u

**Comment by Priyanshu729** [comment]
Pls add hackoberfest label

**Pull request merged**